x/perp/v2/keeper/liquidate.go

```diff
@@ -18,44 +19,51 @@ import (
 func (k Keeper) MultiLiquidate(
 	ctx sdk.Context, liquidator sdk.AccAddress, liquidationRequests []*types.MsgMultiLiquidate_Liquidation,
 ) ([]*types.MsgMultiLiquidateResponse_LiquidationResponse, error) {
-	resp := make([]*types.MsgMultiLiquidateResponse_LiquidationResponse, len(liquidationRequests))
+	resps := make([]*types.MsgMultiLiquidateResponse_LiquidationResponse, len(liquidationRequests))
 
 	var allFailed bool = true
 
-	for i, req := range liquidationRequests {
-		cachedCtx, commit := ctx.CacheContext()
+	for reqIdx, req := range liquidationRequests {
+		// NOTE: writeCachedCtx (1) emits the events on the EventManager of the
+		// cachedCtx. and (2) writes that context to the commit multi store.
+
+		// cachedCtx, writeCachedCtx := ctx.CacheContext()
```

Comment:
We don't want to use cached ctx anymore?

Reply:
Ah, good catch. I forgot to delete these comments. 

Removing the `ctx.CacheContext` stuff was the fix for the duplicate event issue that Oleg was encountering. 

I didn't really understand why we used the cached context before when I read this function. The event manager is a pointer that gets mutated throughout the execution, so it makes sense to append the failed liquidate messages onto it just like the successes.